src/site/content/en/blog/text-fragments/index.md

```diff
@@ -0,0 +1,414 @@
+
+
+### Mixing element and text fragments
+
+Traditional element fragments can be combined with text fragments.
+It is perfectly fine to have both in the same URL, for example,
+to provide a meaningful fallback for browsers that do not support Text Fragments yet.
+The URL
+<a href="https://blog.chromium.org/2019/12/chrome-80-content-indexing-es-modules.html#HTML1:~:text=Give%20us%20feedback%20in%20our%20Product%20Forums."><code>https://blog.chromium.org/2019/12/chrome-80-content-indexing-es-modules.html<strong>#HTML1:~:text=Give%20us%20feedback%20in%20our%20Product%20Forums.</strong></code></a>
+linking to the *Give us feedback in our
+[Product Forums](http://support.google.com/bin/static.py?hl=en&page=portal_groups.cs)* section
+contains both an element fragment (`HTML1`), as well as a text fragment
+(`text=Give%20us%20feedback%20in%20our%20Product%20Forums.`):
+
+<figure class="w-figure">
+  <img src="text-feedback.png" alt="" class="w-screenshot">
+  <figcaption class="w-figcaption">Linking with both element fragment and text fragment.</figcaption>
+</figure>
+
+### The fragment directive
+
+So far I have only quickly glanced over one important addition without actually explaining it:
+the fragment directive `:~:`.
+To avoid compatibility issues with the usage of existing URL element fragments as shown above,
+the [Text Fragments specification](https://wicg.github.io/ScrollToTextFragment/)
+introduces the fragment directive.
+The fragment directive is a portion of the URL fragment delimited by the code sequence `:~:`.
+It is reserved for user agent instructions, such as `text=`,
+and is stripped from the URL during loading so that author scripts cannot directly interact with it.
+User agent instructions are also called *directives*.
+In the concrete case, `text=` is therefore called a *text directive*.
+
+### Feature detectability
+
+Support for Text Fragments is feature-detectable
+by checking for the existence of the `fragmentDirective` in the `Location` interface.
+Unlike the other properties like `hash` that I have mentioned above,
+`fragmentDirective` does not return any value and is read-only.
+The fragment directive is a mechanism for URLs to specify instructions
+directed to the browser rather than the document.
+It is meant to avoid direct interaction with author script,
+so that future user agent instructions can be added without fear
+of introducing breaking changes to existing content.
+One potential example of such future additions could be translation hints.
+
+```js
+if ('fragmentDirective' in window.location) {
+  // Text Fragments is supported.
+}
+```
+
+### Polyfillability
+
+The Text Fragments feature can be polyfilled to some extent.
+There is [work in progress](https://github.com/tomayac/text-fragments-polyfill/)
+to create an extension for browsers that do not support Text Fragments natively
+where the functionality is implemented in JavaScript.
+
+### Security
```

Comment:
It should be noted somewhere that the text fragment is only apply to the main frame - they won't search for text inside an iframe and an iframe navigation won't invoke a text fragment.

For background, this was restricted to reduce the attack surface - we don't have any known specific attacks and this is something we could consider loosening in the future but we thought the reduced attack surface would simplify things initially and we didn't have a strong use case.

Reply:
Added.